### ISSUE TITLE
[CLOSED] Weight Learning with Arithmetic Negative Prior Crashes

### ISSUE THREAD

**Issue by eriq-augustine** [linqs/psl#156]
<a href="https://github.com/eriq-augustine"><img src="https://avatars0.githubusercontent.com/u/337857?v=4" align="left" width="96" height="96" hspace="10"></img></a> **Issue by [eriq-augustine](https://github.com/eriq-augustine)**
_Sunday Oct 08, 2017 at 11:15 GMT_
_Originally opened as https://github.com/eriq-augustine/psl/issues/109_

----

To reproduce, just take any psl-example (easy lp cli in this example), and change the prior from logical:
```
5:  !Knows(P1,P2) ^2
```
to arithmetic:
```
5:  Knows(P1,P2) = 0.0 ^2
```

We get an exception like this:
```
Unexpected exception!
java.lang.NullPointerException
	at org.linqs.psl.reasoner.admm.ADMMReasoner.changedGroundKernelWeight(ADMMReasoner.java:228)
	at org.linqs.psl.reasoner.admm.ADMMReasoner.changedGroundKernelWeights(ADMMReasoner.java:239)
	at org.linqs.psl.application.learning.weight.maxlikelihood.VotedPerceptron.doLearn(VotedPerceptron.java:287)
	at org.linqs.psl.application.learning.weight.WeightLearningApplication.learn(WeightLearningApplication.java:120)
	at org.linqs.psl.cli.Launcher.run(Launcher.java:241)
	at org.linqs.psl.cli.Launcher.main(Launcher.java:351)
```

**Comment by eriq-augustine** [comment]
<a href="https://github.com/eriq-augustine"><img src="https://avatars0.githubusercontent.com/u/337857?v=4" align="left" width="48" height="48" hspace="10"></img></a> **Comment by [eriq-augustine](https://github.com/eriq-augustine)**
_Tuesday Nov 07, 2017 at 01:50 GMT_

----

This may have been fixed with the changes in `lazy-grounding`.
Make sure that you check on that branch, or wait for the merge into `develop`.

**Comment by eriq-augustine** [comment]
<a href="https://github.com/eriq-augustine"><img src="https://avatars0.githubusercontent.com/u/337857?v=4" align="left" width="48" height="48" hspace="10"></img></a> **Comment by [eriq-augustine](https://github.com/eriq-augustine)**
_Wednesday Nov 08, 2017 at 00:31 GMT_

----

I don't think this was fixed yet.
It looks much harder than at first glance.
I think that the parser is parsing the atom as a SummationAtom instead of a normal QueryAtom.
We will have to go back to the parser.
(It also works on some machines.)

**Comment by eriq-augustine** [comment]
<a href="https://github.com/eriq-augustine"><img src="https://avatars0.githubusercontent.com/u/337857?v=4" align="left" width="48" height="48" hspace="10"></img></a> **Comment by [eriq-augustine](https://github.com/eriq-augustine)**
_Wednesday Nov 08, 2017 at 03:09 GMT_

----

Fixed in f8a8446fb71e82c3383bbed1ead0e6b533da6aec.
